### PR TITLE
[clang-tidy] use for range loops

### PR DESCRIPTION
**Pull request by neheb**
Found with modernize-loop-convert

Signed-off-by: Rosen Penev <rosenp@gmail.com>

### CODE REVIEW

File: src/bt_peer_connection.cpp [hunk 1]
```diff
@@ -3730,22 +3729,22 @@ namespace {
 			// the payload ranges they represent have been sent
 			auto first_to_keep = m_payloads.begin();
 
-			for (auto i = m_payloads.begin(); i != m_payloads.end(); ++i)
+			for (auto& m_payload : m_payloads)
```
Comment (arvidn) [inline src/bt_peer_connection.cpp:3732]: this is not a good name for an element in `m_payloads`

File: src/torrent.cpp [hunk 2]
```diff
@@ -3762,17 +3762,17 @@ namespace {
 
 		// look at all unfinished pieces and add the completed
 		// blocks to our 'done' counter
-		for (auto i = dl_queue.begin(); i != dl_queue.end(); ++i)
+		for (auto i : dl_queue)
```
Comment (arvidn) [inline src/torrent.cpp:3765]: this should be `auto const& i`

File: src/piece_picker.cpp [hunk 3]
```diff
@@ -727,12 +727,11 @@ namespace libtorrent {
 		// and also the number of pieces that have more than that.
 		int integer_part = 0;
 		int fraction_part = 0;
-		for (std::vector<piece_pos>::const_iterator i = m_piece_map.begin()
-			, end(m_piece_map.end()); i != end; ++i)
+		for (auto i : m_piece_map)
```
Comment (arvidn) [inline src/piece_picker.cpp:730]: this should be `auto const& i`, shouldn't it?

**Comment by arvidn** [comment]
https://ci.appveyor.com/project/arvidn/libtorrent/builds/33472331
https://travis-ci.org/github/arvidn/libtorrent/builds/697416252

File: src/upnp.cpp [hunk 4]
```diff
@@ -691,13 +691,13 @@ void upnp::try_map_upnp()
 	TORRENT_ASSERT(is_single_thread());
 	if (m_devices.empty()) return;
 
-	for (auto i = m_devices.begin(), end(m_devices.end()); i != end; ++i)
+	for (auto const& m_device : m_devices)
```
Comment (arvidn) [inline src/upnp.cpp:694]: this is not a good name for a entry in `m_devices`

File: src/piece_picker.cpp (hunk 3 above)
Comment (neheb) [reply src/piece_picker.cpp:730]: No idea. I also ran this through clang’s -Wrange-loop-analysis . Didn’t complain.

File: src/upnp.cpp (hunk 4 above)
Comment (neheb) [reply src/upnp.cpp:694]: device better?

File: src/bt_peer_connection.cpp (hunk 1 above)
Comment (neheb) [reply src/bt_peer_connection.cpp:3732]: payload better?

File: src/piece_picker.cpp (hunk 3 above)
Comment (arvidn) [reply src/piece_picker.cpp:730]: your change will make a copy of the `piece_pos` object. The previous code didn't

File: src/bt_peer_connection.cpp (hunk 1 above)
Comment (arvidn) [reply src/bt_peer_connection.cpp:3732]: I think since the scope is small, a short name is ok. like `p`

Comment (arvidn) [reply src/bt_peer_connection.cpp:3732]: or `i`

File: src/upnp.cpp (hunk 4 above)
Comment (arvidn) [reply src/upnp.cpp:694]: since the scope is small, I think `d` or `i` would be fine

**Comment by neheb** [comment]
Fixed everything.

**Comment by arvidn** [comment]
https://ci.appveyor.com/project/arvidn/libtorrent/builds/33496066/job/i6atnn3sxlh7w1hn#L220
https://ci.appveyor.com/project/arvidn/libtorrent/builds/33496066/job/i6atnn3sxlh7w1hn#L293

**Comment by arvidn** [comment]
https://travis-ci.org/github/arvidn/libtorrent/jobs/697846820#L897
https://travis-ci.org/github/arvidn/libtorrent/jobs/697846820#L846

Does it build for you?

**Pull request closed without merging**